http/transactions_middleware.go

```diff
@@ -0,0 +1,110 @@
+// This file contains an HTTP middleware that associates a transaction to each request.
+package http
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"gitlab.cee.redhat.com/service/uhc-clusters-service/pkg/global"
+	"gitlab.cee.redhat.com/service/uhc-clusters-service/pkg/transaction"
+)
+
+// TransactionResponseWriter is a wrapper for http.ResponseWriter,
+// it attempts to resolve (commit or rollback) the transaction on the context of the request
+// if the transaction failed for some reason it catches the error and returns a 500 response
+// with a clear error indicating the failure.
+type TransactionResponseWriter struct {
```

Comment:
This is going to be used only as in implementation detail of the middleware, so it should be private.